lib/amqp-ts.js

```diff
@@ -301,24 +301,34 @@ var Message = (function () {
         if (routingKey === void 0) { routingKey = ""; }
         // inline function to send the message
         var sendMessage = function () {
-            try {
-                destination._channel.publish(exchange, routingKey, _this.content, _this.properties);
-            }
-            catch (err) {
-                exports.log.log("debug", "Publish error: " + err.message, { module: "amqp-ts" });
-                var destinationName = destination._name;
-                var connection = destination._connection;
-                exports.log.log("debug", "Try to rebuild connection, before Call.", { module: "amqp-ts" });
-                connection._rebuildAll(err).then(function () {
-                    exports.log.log("debug", "Retransmitting message.", { module: "amqp-ts" });
-                    if (destination instanceof Queue) {
-                        connection._queues[destinationName].publish(_this.content, _this.properties);
-                    }
-                    else {
-                        connection._exchanges[destinationName].publish(_this.content, routingKey, _this.properties);
-                    }
-                });
-            }
+            return new Promise((resolve, reject) => {
+                try {
+                    destination._channel.publish(exchange, routingKey, _this.content, _this.properties)
+                        .then(resolve)
+                        .catch((e) => {
+                            throw e;
```

Comment:
Should this be `.catch(reject)` instead?

Reply:
I don't think so.

I implemented it this way because there is still a lot of unknown stuff happening in `publish` that I'm not familiar with.

I wanted to ensure that if an error is thrown somewhere in `publish` that it was caught the same way it was caught before my code was added. This means that we need to rethrow the error if the promise rejects, and it will fall to the `catch` block below to deal with it as it was in the past.

Reply:
Talked offline, there was indeed a problem with this implementation (the `throw e;` would not be caught by the catch block below).

The implementation has since been modified.